### PR TITLE
Finish client GUI

### PR DESCRIPTION
**Pull request by 770grappenmaker**
Title. My first QM contribution, please be gentle!

Additional fixes:
- Move `SubtitlePositioner` (not only GUI logic)
- Remove invalid `PacketReportProvider` mapping (by gradle task)
- Append `website` to wordlist

### CODE REVIEW

**Comment by OroArmor** [comment]
Can you enable actions on your fork of the repository so that I can download the generated diff?

**Comment by 770grappenmaker** [comment]
> Can you enable actions on your fork of the repository so that I can download the generated diff?

Should be enabled now. Since I pushed a new commit as well, it performs a fresh run of the workflows, which I couldn't get it to do with the old commit.

File: mappings/net/minecraft/client/gui/screen/TelemetryDataScreen.mapping [hunk 1]
```diff
@@ -2,21 +2,33 @@ CLASS net/minecraft/unmapped/C_reswafjn net/minecraft/client/gui/screen/Telemetr
...
 	FIELD f_etjnsxlc eventWidget Lnet/minecraft/unmapped/C_kejfmlbi;
 	FIELD f_eurrcfjj savedScroll D
+	FIELD f_gyozyqer TELEMETRY_AVAILABLE Z
 	FIELD f_irgyvksw parent Lnet/minecraft/unmapped/C_wrmtlwqx;
 	FIELD f_jzfoorlv PRIVACY_STATEMENT Lnet/minecraft/unmapped/C_rdaqiwdt;
+	FIELD f_lzxjeugb SHOW_DATA Lnet/minecraft/unmapped/C_rdaqiwdt;
 	FIELD f_mbofpaun DESCRIPTION Lnet/minecraft/unmapped/C_rdaqiwdt;
 	FIELD f_nvyvlvqb OPT_IN_DESCRIPTION Lnet/minecraft/unmapped/C_rdaqiwdt;
+	FIELD f_qjsrbscl SPACING I
+	FIELD f_tvkntoyx descriptionText Lnet/minecraft/unmapped/C_ivtezidq;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/gui/screen/TelemetryDataScreen.mapping]:
```suggestion
	FIELD f_etjnsxlc event Lnet/minecraft/unmapped/C_kejfmlbi;
	FIELD f_eurrcfjj savedScroll D
	FIELD f_gyozyqer TELEMETRY_AVAILABLE Z
	FIELD f_irgyvksw parent Lnet/minecraft/unmapped/C_wrmtlwqx;
	FIELD f_jzfoorlv PRIVACY_STATEMENT Lnet/minecraft/unmapped/C_rdaqiwdt;
	FIELD f_lzxjeugb SHOW_DATA Lnet/minecraft/unmapped/C_rdaqiwdt;
	FIELD f_mbofpaun DESCRIPTION Lnet/minecraft/unmapped/C_rdaqiwdt;
	FIELD f_nvyvlvqb OPT_IN_DESCRIPTION Lnet/minecraft/unmapped/C_rdaqiwdt;
	FIELD f_qjsrbscl SPACING I
	FIELD f_tvkntoyx description Lnet/minecraft/unmapped/C_ivtezidq;
```
makes the widget names consistent and brief

File: mappings/net/minecraft/client/gui/screen/WarningScreen.mapping [hunk 2]
```diff
@@ -1,8 +1,11 @@
 CLASS net/minecraft/unmapped/C_ppzrtmwo net/minecraft/client/gui/screen/WarningScreen
 	FIELD f_fnmhtbpo checkboxText Lnet/minecraft/unmapped/C_rdaqiwdt;
 	FIELD f_gxdebllp confirm Lnet/minecraft/unmapped/C_rdaqiwdt;
+	FIELD f_kcgmkuhq textWidget Lnet/minecraft/unmapped/C_rugqjeey;
+	FIELD f_ledlfprk frame Lnet/minecraft/unmapped/C_ozrdqjbi;
 	FIELD f_pugijbby checkbox Lnet/minecraft/unmapped/C_lvxxsthn;
 	FIELD f_ynpbeaxj multilineMessage Lnet/minecraft/unmapped/C_rdaqiwdt;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/gui/screen/WarningScreen.mapping]:
```suggestion
	FIELD f_kcgmkuhq message Lnet/minecraft/unmapped/C_rugqjeey;
	FIELD f_ledlfprk frame Lnet/minecraft/unmapped/C_ozrdqjbi;
	FIELD f_pugijbby checkbox Lnet/minecraft/unmapped/C_lvxxsthn;
	FIELD f_ynpbeaxj messageText Lnet/minecraft/unmapped/C_rdaqiwdt;
```
makes
`Text messageText`->`OutlinedMultilineTextWidget message`
`Text checkboxText`->`CheckboxWidget checkbox`
consistent and brief

File: mappings/net/minecraft/client/gui/screen/realms/RealmsBackupScreen.mapping [hunk 3]
```diff
@@ -34,9 +43,21 @@ CLASS net/minecraft/unmapped/C_jyywrsnb net/minecraft/client/gui/screen/realms/R
...
 		METHOD m_geqrcqvo updateChangeList (Lnet/minecraft/unmapped/C_iyfhccqr;)V
 			ARG 1 backup
+		METHOD m_hdeduczx (Lnet/minecraft/unmapped/C_buwziidm;)V
+			ARG 1 button
+		METHOD m_igmscjbz (Ljava/util/function/Supplier;)Lnet/minecraft/unmapped/C_npqneive;
+			ARG 1 textSupplier
+		METHOD m_ojxypplh showRestorePrompt ()V
 		METHOD m_sjkwqmgw getMediumDatePresentation (Ljava/util/Date;)Ljava/lang/String;
 			ARG 1 lastModifiedDate
+		METHOD m_vamrbvmo getLastModifiedDateString ()Ljava/lang/String;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/gui/screen/realms/RealmsBackupScreen.mapping]:
```suggestion
		METHOD m_vamrbvmo getFormattedLastModifiedDate ()Ljava/lang/String;
```
a little more descriptive and doesn't put the return type in the name

File: mappings/net/minecraft/client/gui/screen/realms/RealmsMainScreen.mapping [hunk 4]
```diff
@@ -261,18 +309,30 @@ CLASS net/minecraft/unmapped/C_uncfwjlr net/minecraft/client/gui/screen/realms/R
 		METHOD m_uwnplzte drawNotificationCountTexture (Lnet/minecraft/unmapped/C_sedilmty;)V
 	CLASS C_vvkxoqra RealmEntry
 		FIELD f_aznycvbu tooltip Lnet/minecraft/unmapped/C_fasqumtv;
+		FIELD f_ilobelrl ONLINE_PLAYERS Lnet/minecraft/unmapped/C_rdaqiwdt;
 		FIELD f_jfvtejrc SKIN_HEAD_LARGE_WIDTH I
 		FIELD f_qpfilxjr serverData Lnet/minecraft/unmapped/C_pwmwypfb;
+		FIELD f_unxbjyfx PLAYERS_ONLINE_WIDTH I
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/gui/screen/realms/RealmsMainScreen.mapping]:
```suggestion
		FIELD f_unxbjyfx PLAYER_FACE_SIZE I
```
I think that's what this is based on `renderOnlinePlayers`, esp:
```java
PlayerFaceRenderer.draw(graphics, MinecraftClient.getInstance().getSkinProvider().getSkin(list.get(r).profile()), q, p, 9);
```
`9` is the value of the constant, and it's being used for the `size` parameter of `PlayerFaceRenderer::draw`
I also only see `9` used for size/y/height calculations, not x/width calculations.

mojmap is `PLAYERS_ONLINE_SPRITE_SIZE`

File: mappings/net/minecraft/client/gui/widget/list/EntryListWidget.mapping [hunk 5]
```diff
@@ -61,16 +69,21 @@ CLASS net/minecraft/unmapped/C_nmdfcqgq net/minecraft/client/gui/widget/list/Ent
...
 		ARG 2 mouseX
 		ARG 3 mouseY
+	METHOD m_koseqkma canScroll ()Z
 	METHOD m_krleyrsx scroll (I)V
 		ARG 1 amount
 	METHOD m_ldjgvvhr getRowTop (I)I
 		ARG 1 index
+	METHOD m_lfrxnnzk getDefaultScrollbarPositionX ()I
+	METHOD m_mfmjanjh getUnpaddedRowRight ()I
+	METHOD m_mgkibvfk recomputeScrollAmount ()V
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/gui/widget/list/EntryListWidget.mapping]:
```suggestion
	METHOD m_mgkibvfk refreshScrollAmount ()V
```
I think makes it clearer that it sets the scroll amount

File: mappings/net/minecraft/client/gui/widget/list/world/WorldEntryListWidget.mapping [hunk 6]
```diff
@@ -70,8 +70,15 @@ CLASS net/minecraft/unmapped/C_wwgrvynj net/minecraft/client/gui/widget/list/wor
 			ARG 2 worldEntryList
 			ARG 3 summary
 		METHOD m_aehstgkl recreate ()V
+		METHOD m_fijclnvs (Lnet/minecraft/unmapped/C_tmnkpzlg;Lnet/minecraft/unmapped/C_njsjipmy;Ljava/nio/file/Path;Z)V
+			ARG 4 clickedProceed
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/gui/widget/list/world/WorldEntryListWidget.mapping]:
```suggestion
			ARG 4 confirmed
```

File: mappings/net/minecraft/client/gui/widget/list/world/WorldEntryListWidget.mapping [hunk 7]
```diff
@@ -70,8 +70,15 @@ CLASS net/minecraft/unmapped/C_wwgrvynj net/minecraft/client/gui/widget/list/wor
...
+		METHOD m_fyvuyosl (Lnet/minecraft/unmapped/C_bpwagnit$C_unecclfw;Z)V
+			ARG 2 successful
 		METHOD m_hnhktyvf getLevelDisplayName ()Ljava/lang/String;
+		METHOD m_itvygymc canSelect ()Z
 		METHOD m_ldplqrie resolveIconFile ()V
+		METHOD m_lmhxixgl (Z)V
+			ARG 1 clickedDelete
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/gui/widget/list/world/WorldEntryListWidget.mapping]:
```suggestion
			ARG 1 confirmed
```

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]
looks good, mostly just nitpicks

File: mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping [hunk 8]
```diff
@@ -49,13 +50,18 @@ CLASS net/minecraft/unmapped/C_uedmlrsy net/minecraft/client/gui/hud/chat/ChatHu
...
+		ARG 1 message
 	METHOD m_itcpovxa toChatLineX (D)D
 		ARG 1 x
+	METHOD m_iudfmgeq toChatState ()Lnet/minecraft/unmapped/C_uedmlrsy$C_frlehikz;
 	METHOD m_kyfuayab (ILnet/minecraft/unmapped/C_uedmlrsy$C_hrdqyvbn;)Z
 		ARG 2 message
 	METHOD m_lqhjyrrj createDelayedMessageDeletion (Lnet/minecraft/unmapped/C_shphkjbi;)Lnet/minecraft/unmapped/C_uedmlrsy$C_hrdqyvbn;
 		ARG 1 signature
 	METHOD m_maliiwol updateDeletionQueue ()V
+	METHOD m_mqmrcyqk addVisibleMessage (Lnet/minecraft/unmapped/C_skejqrts;)V
```
Comment (eristhea) [inline mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping]:
```suggestion
	METHOD m_mqmrcyqk showMessage (Lnet/minecraft/unmapped/C_skejqrts;)V
```

maybe? i think addVisible is kinda naming on an impl detail

File: mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping [hunk 9]
```diff
@@ -32,6 +32,7 @@ CLASS net/minecraft/unmapped/C_uedmlrsy net/minecraft/client/gui/hud/chat/ChatHu
 	METHOD m_bzentyiw addMessage (Lnet/minecraft/unmapped/C_rdaqiwdt;)V
 		ARG 1 message
 	METHOD m_csgqmavu tickDeletionQueue ()V
+	METHOD m_doerxbdt resetVisibleMessages ()V
```
Comment (eristhea) [inline mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping]:
```suggestion
	METHOD m_doerxbdt refreshVisibleMessages ()V
```

i feel like `reset` is a weird word to go for here, maybe `refresh` instead?

File: mappings/net/minecraft/client/gui/screen/realms/RealmsMainScreen.mapping [hunk 10]
```diff
@@ -211,32 +243,48 @@ CLASS net/minecraft/unmapped/C_uncfwjlr net/minecraft/client/gui/screen/realms/R
 			ARG 1 onPress
 			ARG 2 message
 	CLASS C_hsrdmghw AbstractEntry
+		FIELD f_fzyoaley SPACING_Y I
```
Comment (eristhea) [inline mappings/net/minecraft/client/gui/screen/realms/RealmsMainScreen.mapping]:
```suggestion
		FIELD f_fzyoaley Y_SPACING I
```

natural word order

File: mappings/net/minecraft/client/gui/screen/realms/RealmsMainScreen.mapping [hunk 11]
```diff
@@ -211,32 +243,48 @@ CLASS net/minecraft/unmapped/C_uncfwjlr net/minecraft/client/gui/screen/realms/R
...
 		FIELD f_pielkvwx STATUS_LIGHT_WIDTH I
+		FIELD f_tjokbasd SPACING_X I
```
Comment (eristhea) [inline mappings/net/minecraft/client/gui/screen/realms/RealmsMainScreen.mapping]:
```suggestion
		FIELD f_tjokbasd X_SPACING I
```

File: mappings/net/minecraft/client/gui/screen/world/CreateWorldScreen.mapping [hunk 12]
```diff
@@ -65,27 +76,44 @@ CLASS net/minecraft/unmapped/C_ibkofvzc net/minecraft/client/gui/screen/world/Cr
...
+	METHOD m_uvqcxsuo (Ljava/nio/file/Path;)Z
+		ARG 1 childPath
 	METHOD m_vfaqeuwk (Lnet/minecraft/unmapped/C_ibkofvzc;Lnet/minecraft/unmapped/C_fxiqpxaf;)V
 		ARG 1 widget
+	METHOD m_vjzwlndv (Ljava/util/function/Consumer;Ljava/lang/Void;Ljava/lang/Throwable;)Ljava/lang/Object;
+		ARG 2 result
+		ARG 3 error
 	METHOD m_xdqbchlc createWorld ()V
+	METHOD m_ybclynik (Lorg/apache/commons/lang3/mutable/MutableObject;Ljava/nio/file/Path;Ljava/nio/file/Path;)V
+		ARG 2 childPAth
```
Comment (eristhea) [inline mappings/net/minecraft/client/gui/screen/world/CreateWorldScreen.mapping]:
```suggestion
		ARG 2 childPath
```
typo

File: mappings/net/minecraft/client/gui/widget/CheckboxWidget.mapping [hunk 13]
```diff
@@ -1,19 +1,71 @@
 CLASS net/minecraft/unmapped/C_lvxxsthn net/minecraft/client/gui/widget/CheckboxWidget
+	FIELD f_cnayjurh listener Lnet/minecraft/unmapped/C_lvxxsthn$C_npauekvc;
 	FIELD f_epfkmuwh CHECKBOX_HIGHLIGHTED Lnet/minecraft/unmapped/C_ncpywfca;
+	FIELD f_fiwvdbpa PADDING_X I
```
Comment (eristhea) [inline mappings/net/minecraft/client/gui/widget/CheckboxWidget.mapping]:
```suggestion
	FIELD f_fiwvdbpa X_PADDING I
```
natural order, again

File: mappings/net/minecraft/client/gui/widget/CheckboxWidget.mapping [hunk 14]
```diff
@@ -1,19 +1,71 @@
...
+		METHOD m_piftuvee maxWidth (I)Lnet/minecraft/unmapped/C_lvxxsthn$C_beiircfb;
+			ARG 1 maxWidth
+		METHOD m_rygxmxec position (II)Lnet/minecraft/unmapped/C_lvxxsthn$C_beiircfb;
+			ARG 1 x
+			ARG 2 y
+		METHOD m_zgugmcut checkListener (Lnet/minecraft/unmapped/C_lvxxsthn$C_npauekvc;)Lnet/minecraft/unmapped/C_lvxxsthn$C_beiircfb;
+			ARG 1 listener
+		METHOD m_zjwvugfl build ()Lnet/minecraft/unmapped/C_lvxxsthn;
+	CLASS C_npauekvc CheckListener
+		FIELD f_mpjhkppa NOOP Lnet/minecraft/unmapped/C_lvxxsthn$C_npauekvc;
```
Comment (eristhea) [inline mappings/net/minecraft/client/gui/widget/CheckboxWidget.mapping]: looks like these are inconsistent, i prefer NO_OP over NOOP since i think it's more understandable if you haven't heard the phrase no-op before but i'd like to know what others think

File: mappings/net/minecraft/client/gui/widget/list/EntryListWidget.mapping [hunk 15]
```diff
@@ -97,6 +122,9 @@ CLASS net/minecraft/unmapped/C_nmdfcqgq net/minecraft/client/gui/widget/list/Ent
 	METHOD m_ulzatvjk getEntryAtPosition (DD)Lnet/minecraft/unmapped/C_nmdfcqgq$C_rzayyeiy;
 		ARG 1 x
 		ARG 3 y
+	METHOD m_umtazozw getScrollbarPaddingX ()I
```
Comment (eristhea) [inline mappings/net/minecraft/client/gui/widget/list/EntryListWidget.mapping]:
```suggestion
	METHOD m_umtazozw getScrollbarXPadding ()I
```
the natural word order warrior

File: mappings/net/minecraft/client/sound/SoundTransform.mapping [hunk 16]
```diff
@@ -0,0 +1,12 @@
+CLASS net/minecraft/unmapped/C_cfcnvldd net/minecraft/client/sound/SoundTransform
+	COMMENT Represents the camera state of the player. This is used in two ways:
+	COMMENT - Passing the camera transform to OpenAL for positional audio
+	COMMENT - Computing the direction a sound came from for the subtitles hud
```
Comment (eristhea) [inline mappings/net/minecraft/client/sound/SoundTransform.mapping]:
```suggestion
	COMMENT - Computing the direction a sound came from for the subtitle display
```
i think this is clearer

**Review by eristhea — COMMENTED** [review]
looks great, thank you so much for the huge PR! just a few suggestions

File: mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping (hunk 8 above)
Comment (770grappenmaker) [reply mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping]: I considered `showMessage` but I thought it was too different from `addMessage`, whereas it kind of does the same thing. `showMessage` imo implies rendering it, which admittedly it will but not _this frame_.

File: mappings/net/minecraft/client/gui/widget/CheckboxWidget.mapping (hunk 14 above)
Comment (770grappenmaker) [reply mappings/net/minecraft/client/gui/widget/CheckboxWidget.mapping]: Judging by other QM names, both are used, so I honestly wouldn't know which to prefer.

File: mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping (hunk 8 above)
Comment (770grappenmaker) [reply mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping]: For now won't resolve this conversation in order to discuss further if necessary.

**Comment by 770grappenmaker** [comment]
Applied some of your suggestions, let me know what you think!

File: mappings/net/minecraft/client/gui/WorldGenerationProgressTracker.mapping [hunk 17]
```diff
@@ -6,9 +6,14 @@ CLASS net/minecraft/unmapped/C_qtaaqxsg net/minecraft/client/gui/WorldGeneration
...
+	METHOD m_bpcnlyan createWithTotalRadius (I)Lnet/minecraft/unmapped/C_qtaaqxsg;
+		ARG 0 spawnChunkRadius
+	METHOD m_inajsbju createWithRadius (I)Lnet/minecraft/unmapped/C_qtaaqxsg;
+		ARG 0 spawnChunkRadius
 	METHOD m_nmznsyyi getChunkStatus (II)Lnet/minecraft/unmapped/C_kogtzhzt;
 		ARG 1 x
 		ARG 2 z
 	METHOD m_npoofhgh getCenterSize ()I
 	METHOD m_obkavrre getSize ()I
+	METHOD m_ryrzyybt createWithoutSpawnRadius ()Lnet/minecraft/unmapped/C_qtaaqxsg;
```
Comment (770grappenmaker) [inline mappings/net/minecraft/client/gui/WorldGenerationProgressTracker.mapping:18]: Not sure if these are descriptive enough, thoughts?

File: mappings/net/minecraft/client/gui/screen/realms/RealmsDialogs.mapping [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+	FIELD f_sxpczgxs INFO_COLOR I
+	METHOD m_ghrkomsd createInfo (Lnet/minecraft/unmapped/C_wrmtlwqx;Lnet/minecraft/unmapped/C_rdaqiwdt;Ljava/util/function/Consumer;)Lnet/minecraft/unmapped/C_pzklyvtm;
+		ARG 0 screen
+		ARG 1 text
+		ARG 2 consumer
+	METHOD m_khmvieru createWarning (Lnet/minecraft/unmapped/C_wrmtlwqx;Lnet/minecraft/unmapped/C_rdaqiwdt;Ljava/util/function/Consumer;)Lnet/minecraft/unmapped/C_pzklyvtm;
+		ARG 0 screen
+		ARG 1 text
+		ARG 2 consumer
+	METHOD m_umgginml createWarningMessage (Lnet/minecraft/unmapped/C_wrmtlwqx;Lnet/minecraft/unmapped/C_rdaqiwdt;Ljava/util/function/Consumer;)Lnet/minecraft/unmapped/C_pzklyvtm;
```
Comment (770grappenmaker) [inline mappings/net/minecraft/client/gui/screen/realms/RealmsDialogs.mapping:13]: Are these names good enough?

File: mappings/net/minecraft/client/gui/hud/debug/DebugHud.mapping [hunk 19]
```diff
@@ -4,6 +4,7 @@ CLASS net/minecraft/unmapped/C_epxplegp net/minecraft/client/gui/hud/debug/Debug
 	FIELD f_demcwxlr client Lnet/minecraft/unmapped/C_ayfeobid;
 	FIELD f_fjhfdzhz HEIGHT_MAP_TYPES Ljava/util/Map;
 	FIELD f_hdjainem fluidHit Lnet/minecraft/unmapped/C_eetoegzn;
+	FIELD f_jvalscle serverPerformanceSampleLogs Ljava/util/Map;
```
Comment (770grappenmaker) [inline mappings/net/minecraft/client/gui/hud/debug/DebugHud.mapping:7]: While mapping these debug hud related things, I've also got some changes locally to everything related to performance sampling and logging, should I make a separate PR for that, or should I cherry-pick those onto this PR?

Comment (770grappenmaker) [reply mappings/net/minecraft/client/gui/hud/debug/DebugHud.mapping:7]: For example, both GUI code and server-related code reference `net/minecraft/client/gui/hud/debug/PerformanceSampleLog`, so the package is probably wrong. I've also mapped all of the related types and changed some names, which I believe doesn't really fit into the nature of this PR, focusing mostly on adding mappings.

File: mappings/net/minecraft/client/gui/screen/CreditsScreen.mapping [hunk 20]
```diff
@@ -4,15 +4,18 @@ CLASS net/minecraft/unmapped/C_emwfaans net/minecraft/client/gui/screen/CreditsS
 	FIELD f_cuxehlzw SEPARATOR_LINE Lnet/minecraft/unmapped/C_rdaqiwdt;
 	FIELD f_cxzuezzk CENTERED_LINE_PREFIX Ljava/lang/String;
 	FIELD f_dakivsbd logoRenderer Lnet/minecraft/unmapped/C_siglwpps;
+	FIELD f_ewbhrgtm POST_CREDITS_TEXT Lnet/minecraft/unmapped/C_ncpywfca;
 	FIELD f_ftsdfuxo centeredLines Lit/unimi/dsi/fastutil/ints/IntSet;
 	FIELD f_fyszbzgh speed F
+	FIELD f_hefuvaro END_POEM Lnet/minecraft/unmapped/C_ncpywfca;
```
Comment (770grappenmaker) [inline mappings/net/minecraft/client/gui/screen/CreditsScreen.mapping:10]: This name might be a bit too creative since it doesn't match the string `"texts/end.txt"`, but it feels more recognizable than `END_TEXT`

**Review by 770grappenmaker — COMMENTED** [review]
Some thoughts of mine

File: mappings/net/minecraft/client/gui/screen/CreditsScreen.mapping (hunk 20 above)
Comment (770grappenmaker) [reply mappings/net/minecraft/client/gui/screen/CreditsScreen.mapping:10]: mojmap is `END_POEM_LOCATION`, so I guess not too creative after all

File: mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping (hunk 8 above)
Comment (OroArmor) [reply mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping]: Maybe `queueMessage`?

File: mappings/net/minecraft/client/gui/widget/CheckboxWidget.mapping (hunk 14 above)
Comment (OroArmor) [reply mappings/net/minecraft/client/gui/widget/CheckboxWidget.mapping]: I think standardizing around no op is better. We are make an issue and update the conventions

File: mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping (hunk 8 above)
Comment (770grappenmaker) [reply mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping]: Might be better, though the visible/display/show element is missing, and it isn't too far off from `addMessage`. mojmap is `addMessageToDisplayQueue` so honestly in the right direction. Unsure what to do.

Comment (OroArmor) [reply mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping]: `queueVisbileMessage`?

Comment (eristhea) [reply mappings/net/minecraft/client/gui/hud/chat/ChatHud.mapping]: I like that

File: mappings/net/minecraft/client/gui/screen/realms/RealmsPlayerScreen.mapping [hunk 21]
```diff
@@ -16,12 +22,41 @@ CLASS net/minecraft/unmapped/C_owmmvzoi net/minecraft/client/gui/screen/realms/R
...
+			ARG 1 index
+		METHOD m_hvkguzat (Lnet/minecraft/unmapped/C_uckflzvp;Ljava/util/function/Supplier;)Lnet/minecraft/unmapped/C_npqneive;
+			ARG 1 textSupplier
+		METHOD m_jagblmvl (Lnet/minecraft/unmapped/C_uckflzvp;Ljava/util/function/Supplier;)Lnet/minecraft/unmapped/C_npqneive;
+			ARG 1 textSupplier
 		METHOD m_lcrgzomf (ILnet/minecraft/unmapped/C_buwziidm;)V
 			ARG 2 button
+		METHOD m_lerxpmdr updateOperatorButtons ()V
+		METHOD m_ntvsespc getOperatorButton ()Lnet/minecraft/unmapped/C_buwziidm;
+		METHOD m_nycgvaso uninvitePlayer (I)V
```
Comment (OroArmor) [inline mappings/net/minecraft/client/gui/screen/realms/RealmsPlayerScreen.mapping:49]:
```suggestion
		METHOD m_nycgvaso removePlayer (I)V
```

File: mappings/net/minecraft/util/ServerLinks.mapping [hunk 22]
```diff
@@ -0,0 +1,63 @@
+CLASS net/minecraft/unmapped/C_ouodvkzu net/minecraft/util/ServerLinks
+	FIELD f_ctwoeegm entries Ljava/util/List;
+	FIELD f_cvvvvdle EMPTY Lnet/minecraft/unmapped/C_ouodvkzu;
+	FIELD f_hskldvmh STRING_ENTRIES_CODEC Lnet/minecraft/unmapped/C_qsrmwluu;
+	FIELD f_rwqplgxr URI_ENTRY_CODEC Lnet/minecraft/unmapped/C_qsrmwluu;
```
Comment (OroArmor) [inline mappings/net/minecraft/util/ServerLinks.mapping]: these should probably end with `PACKET_CODEC` instead of `CODEC`

**Review by OroArmor — CHANGES_REQUESTED** [review]
Some small things

File: mappings/net/minecraft/client/gui/screen/realms/RealmsPlayerScreen.mapping (hunk 21 above)
Comment (770grappenmaker) [reply mappings/net/minecraft/client/gui/screen/realms/RealmsPlayerScreen.mapping:49]: I chose `uninvitePlayer` because of the strings `"Couldn't uninvite user"` and `"mco.configure.world.uninvite.player"` and the delegation to `RealmsClient#uninvite`. I'm not sure if the code overlaps with the code to remove players that are already members. mojmap is `uninvite`

Comment (OroArmor) [reply mappings/net/minecraft/client/gui/screen/realms/RealmsPlayerScreen.mapping:49]: Ah I see. I was looking at where it was called from the buttons, and that all has `remove` associated with it.

File: mappings/net/minecraft/client/gui/WorldGenerationProgressTracker.mapping (hunk 17 above)
Comment (eristhea) [reply mappings/net/minecraft/client/gui/WorldGenerationProgressTracker.mapping:18]: these look good to me! i like that you delegated mentioning spawn chunks to the parameters

File: mappings/net/minecraft/client/gui/hud/debug/DebugHud.mapping (hunk 19 above)
Comment (eristhea) [reply mappings/net/minecraft/client/gui/hud/debug/DebugHud.mapping:7]: I would put those refactors in another PR, this one is fairly huge already

File: mappings/net/minecraft/client/gui/screen/CreditsScreen.mapping (hunk 20 above)
Comment (eristhea) [reply mappings/net/minecraft/client/gui/screen/CreditsScreen.mapping:10]: the current name is good

**Review by eristhea — APPROVED** [review]

**Comment by 770grappenmaker** [comment]
Did not mean to rerequest @ix0rai for review (screw you github for moving the UI unexpectedly)

**Review by eristhea — APPROVED** [review]
double approved

**Review by OroArmor — APPROVED** [review]

**Review by supersaiyansubtlety — APPROVED** [review]

**Comment by 770grappenmaker** [comment]
When would this be eligible for merge?

**Comment by supersaiyansubtlety** [comment]
whoops, I was supposed to start fcp when I approved, mb
